src/crab/ebpf_domain.cpp

```diff
@@ -2154,8 +2154,61 @@ void ebpf_domain_t::do_mem_store(const Mem& b, Type val_type, SValue val_svalue,
     });
 }
 
-void ebpf_domain_t::operator()(const LockAdd& a) {
-    // nothing to do here
+void ebpf_domain_t::operator()(const Atomic& a) {
+    if (m_inv.is_bottom())
+        return;
+    if (!m_inv.entail(type_is_pointer(reg_pack(a.access.basereg))) ||
```

Comment:
I believe we can return whenever it's not a stack access.

Reply:
Done

Reply:
I just tried that and it caused the "atomic 32-bit ADD and fetch shared" YAML test to fail.  When type is shared, we still need to havoc the src register when FETCH is set, which is done at line 2195.

Reply:
Can we do that immediately and return?

Reply:
Done